daemon/inertiad/crypto/key_derivation.go

```diff
@@ -0,0 +1,34 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const (
+	// KeyDerivationSaltLength is the length of the salt in bytes
+	KeyDerivationSaltLength = 8
+	// KeyDerivationKeyLength is the length of the key derived in bytes
+	KeyDerivationKeyLength  = 32
+	keyDerivationIterations = 10000
+)
+
+// GenerateSalt returns a random hex encoded salt for KD algorithm
+func GenerateSalt() []byte {
+	salt := make([]byte, KeyDerivationSaltLength)
+	rand.Read(salt)
+	return salt
+}
+
+// DeriveKey derives an AES encryption key based on salt + user's
+// password using PBKDF2 with HMAC-SHA1
```

Comment:
I think you meant to say `HMAC-SHA256` here, since that's what you're using in the function.

Reply:
Are you sure? This is Tristan's work but look at line 15 in totp.go